src/components/datepicker/utils.ts

```diff
@@ -32,9 +32,9 @@ export const isAllowedDate = (date: Date, disabledDays?: AllowedDisabledDays) =>
   if (disabledDays instanceof Date) {
     return dateTime !== getDateTimeAtStartOfDay(disabledDays);
   }
-  if (Array.isArray(disabledDays) && disabledDays.length > 0) {
+  if (Array.isArray(disabledDays) && disabledDays.length > 0 && disabledDays[0] instanceof Date) {
     return !disabledDays.some((disabledDay) => {
-      return dateTime === getDateTimeAtStartOfDay(disabledDay);
+      return disabledDay instanceof Date ? dateTime === getDateTimeAtStartOfDay(disabledDay) : false;
```

Comment:
I guess it doesn't make sense to update a disabled test, but can you add a comment to it to remind us to update it for this case?

Reply:
done :)